### PR TITLE
New C example & build tx C-API

### PR DESCRIPTION
**Pull request by moshababo**
Closing https://github.com/spacemeshos/svm/issues/90 & https://github.com/spacemeshos/svm/issues/79 .
The two commits are grouped because they test one another. 

https://github.com/spacemeshos/svm/commit/fcc60e1458143fc7acce201cb2892d551c89af55 (https://github.com/spacemeshos/svm/issues/90):
* Define 3 new c-api methods
* Define convertion from `svm_value` (C-compatible) into `WasmValue` (Rust)
* Define convertion from `svm_value_array` (C-compatible) into `Vec<WasmValue>>` (Rust)

https://github.com/spacemeshos/svm/commit/dbb21f12438966e302e40ff546145c06db4d10d7 (https://github.com/spacemeshos/svm/issues/79):
* New API usage
* Code cleanup
* Remove `wapm` dependency from the build
* Change the WebAssembly app: use storage and introduce separation between storage and host calls

### CODE REVIEW

File: crates/svm-runtime-c-api/src/value.rs [hunk 1]
```diff
@@ -110,6 +137,30 @@ impl From<WasmValue> for svm_value {
...
+/// use svm_runtime_c_api::{svm_value, svm_value_type};
+///
+/// let wasm_val: WasmValue = svm_value { ty: svm_value_type::SVM_I32, i32_val: 10, i64_val: 0 }.into();
+/// assert_eq!(wasm_val, WasmValue::I32(10));
+///
+/// let wasm_val: WasmValue = svm_value { ty: svm_value_type::SVM_I64, i32_val: 0, i64_val: 20 }.into();
+/// assert_eq!(wasm_val, WasmValue::I64(20));
+///
+/// ```
+impl Into<WasmValue> for svm_value {
```
Comment (YaronWittenstein) [inline crates/svm-runtime-c-api/src/value.rs:155]: AFAIK the convention is to prefer `From` over `Into`.
can you please change to 

```
impl From<svm_value> for WasmValue {
  ...
}
```

File: crates/svm-runtime-c-api/src/value.rs [hunk 2]
```diff
@@ -55,6 +55,33 @@ impl From<Vec<WasmValue>> for svm_value_array {
...
+/// let (ptr, len, _cap) = values.into_raw_parts();
+/// let arr = svm_value_array { values: ptr, length: len as u64 };
+///
+/// let vec: Vec<WasmValue> = arr.into();
+/// assert_eq!(vec.len(), 2);
+/// assert_eq!(vec[0], WasmValue::I32(10));
+/// assert_eq!(vec[1], WasmValue::I64(20));
+/// ```
+///
+impl Into<Vec<WasmValue>> for svm_value_array {
```
Comment (YaronWittenstein) [inline crates/svm-runtime-c-api/src/value.rs:78]: same. please change to:

```rust
impl From<svm_value_array> for Vec<WasmValue> {
  ...
}
```

File: examples/c/host_ctx.h [hunk 3]
```diff
@@ -1,5 +1,5 @@
-#ifndef HOST_CTX_H
-#define HOST_CTX_H
+#ifndef PAINTERS_H
```
Comment (YaronWittenstein) [inline examples/c/host_ctx.h:1]: why `PAINTERS_H` ?

File: examples/c/counter.c [hunk 4]
```diff
@@ -114,284 +60,277 @@ void* imports_build() {
...
+
+  // Extract: gas used.
+  uint64_t gas_used;
+  res = svm_exec_receipt_gas(&gas_used, receipt, &err);
+  assert_svm_result(res, err, "svm_app_receipt_gas");
+
+  // Reclaim resources.
+  free(host_ctx.bytes);
+
+  return (exec_app_result_t) {
```
Comment (YaronWittenstein) [inline examples/c/counter.c:223]: I'm not sure the surrounded braces are required here.

File: examples/c/host_ctx.h (hunk 3 above)
Comment (moshababo) [reply examples/c/host_ctx.h:1]: Sorry, my bad.

File: examples/c/counter.c (hunk 4 above)
Comment (moshababo) [reply examples/c/counter.c:223]: I got compilation error without it:
```
unexpected type name 'exec_app_result_t': expected expression
```

**Pull request merged**